src/Tribe/Date_Utils.php

```diff
@@ -1271,7 +1271,19 @@ public static function build_date_object( $datetime = 'now', $timezone = null, $
 		 *              like `strtotime`, or not.
 		 */
 		public static function is_valid_date( $date ) {
-			return self::build_date_object( $date, null, false ) instanceof DateTime;
+			static $cache_var_name = __FUNCTION__;
+
+			$cache_date_check = tribe_get_var( $cache_var_name, [] );
+
+			if ( isset( $cache_date_check[ $date ] ) ) {
+				return $cache_date_check[ $date ];
+			}
+
+			$cache_date_check[ $date ] = self::build_date_object( $date, null, false ) instanceof DateTime;
```

Comment:
Might want to change to an interface instead so in the future if we want to change to `DateTimeImmutable` it might be simpler.

```suggestion
			$cache_date_check[ $date ] = self::build_date_object( $date, null, false ) instanceof DateTimeInterface;
```

It can be replaced with a Carbon one as well, nothing harmful but provides more flexibility along the road.